Azure/azure-cli
# {Cloud} Fix #24946: Fix the bug that config directory may not exist when getting known clouds

**Related command**

`az cloud`

**Description**

Fix #24946 
An error occurred when the environment variable `ARM_CLOUD_METADATA_URL` was set and the `~/.azure` directory was missing. The PR aims to fix the bug that config directory may not exist when executing the function `get_known_clouds`.

**Testing Guide**

Set the environment variable `ARM_CLOUD_METADATA_URL`, remove the `~/.azure` folder and then run any `az` command. No error is expected to happen now.

**History Notes**


---

This checklist is used to make sure that common guidelines for a pull request are followed.
